### PR TITLE
fix: added code to convert schema pointers within a fully resolved swagger

### PR DESCRIPTION
**Pull request by bonnyr**
fix: added code to convert schema pointers within a fully resolved swagger definition back to local references, used when converting a swagger loader into JSON

This generates a fully merged schema definition with all external references, if any, converted to local references.
Moreover, this helps with attempting to convert swagger definitions that contain recursive references and caused the marshalling to blow up with a stack overflow error

This code path is optional and must be called explicitly (FixRefsAndMarshalJSON). Also note that schema references resolved from external schemas need to have their `Ref` property reset

**TODO**
This only currently addressing schemas. The same code can be made to address other references (eg. paths, examples etc)

### CODE REVIEW

File: openapi3/refs.go [hunk 1]
```diff
@@ -79,7 +83,8 @@ func (value HeaderRef) GetRef() string { return value.Ref }
 func (value *HeaderRef) ClearRef()     { value.Ref = "" }
 
 func (value *HeaderRef) MarshalJSON() ([]byte, error) {
-	return jsoninfo.MarshalRef(value.Ref, value.Value)
+	return marshalRef(value.Value, value.Ref)
+	//return jsoninfo.MarshalRef(value.Ref, value.Value)
```
Comment (weberr13) [inline openapi3/refs.go]: comment?

File: openapi3/refs.go [hunk 2]
```diff
@@ -7,6 +7,10 @@ import (
 	"github.com/getkin/kin-openapi/jsoninfo"
 )
 
+func marshalRef(v interface{}, r string) ([]byte, error) {
```
Comment (weberr13) [inline openapi3/refs.go:10]: what is the purpose of this function?

File: openapi3/refs.go (hunk 1 above)
Comment (bonnyr) [reply openapi3/refs.go]: oops, i unified all of these to call a local function to help address the JSONising, but have forgotten this comment somehow

File: openapi3/refs.go (hunk 2 above)
Comment (bonnyr) [reply openapi3/refs.go:10]: see above comment (initial version had a store/lookup mechanism but it was not sufficient. This is still a place holder for future work (although it's not clear to me yet how to go about doing this)

Comment (weberr13) [reply openapi3/refs.go:10]: ok, maybe make a TODO comment to this effect so I can keep track.

**Pull request merged**